# A Harmonized Atlas of Spinal Cord Cell Types and Their Computational Classification

Daniel E. Russ<sup>1,\*</sup>, Ryan B. Patterson-Cross<sup>2,\*</sup>, Li Li<sup>2</sup>, Stephanie C. Koch<sup>3</sup>, Kaya J.E. Matson<sup>2</sup>, Ariel J. Levine<sup>2,#</sup>

_1_ Division of Cancer Epidemiology and Genetics, Data Science Research Group, National Cancer Institute, NIH, Rockville, MD, USA

_2_ Spinal Circuits and Plasticity Unit, National Institute of Neurological Disorders and Stroke, NIH, Bethesda, MD, USA

_3_ Department of Neuroscience, Physiology and Pharmacology, Division of Biosciences, University College of London, London, UK

_\*_ Equal Contribution

_\#_ Corresponding Author

## ABSTRACT

Single cell sequencing is transforming many fields of science but the vast amount of data it creates has the potential to both illuminate and obscure underlying biology. To harness the exciting potential of single cell data for the study of the mouse spinal cord, we have created a harmonized atlas of spinal cord transcriptomic cell types that unifies six independent and disparate studies into one common analysis. With the power of this large and diverse dataset, we reveal spinal cord cell type organization, validate a combinatorial set of markers for in-tissue spatial gene expression analysis, and optimize the computational classification of spinal cord cell types based on transcriptomic data. This work provides a comprehensive resource with unprecedented resolution of spinal cord cell types and charts a path forward for how to utilize transcriptomic data to expand our knowledge of spinal cord biology.
